code/game/objects/items/RPD.dm

```diff
@@ -624,7 +624,8 @@ GLOBAL_LIST_INIT(transit_tube_recipes, list(
 
 						if(queued_p_flipped)
 							tube.setDir(turn(queued_p_dir, 45))
-							tube.SimpleRotateFlip()
+							var/datum/component/simple_rotation/rotcomp = tube.GetComponent(/datum/component/simple_rotation)
+							rotcomp?.Rotate(user, ROTATION_FLIP)
```

Comment:
If memory serves me right, reckless usage of the GetComponent component is discouraged, yet I'm hesitant to make it a signal. Given how particular this is, I'd rather not twiddle around this `pre_attack()` call that much.

Reply:
any reason this can't just set dir directly?

Reply:
I'll go check if it's safe to do so. Because the Rotate() proc does more than setting the object's dir, I cannot give you a yes or no answer right away.